### DISCUSSION TITLE
docker.io/rabbitmq:3-management-alpine container image list cannot be imported into quay.io

### DISCUSSION THREAD

**Discussion by mh21** [docker-library/rabbitmq#660] [General]
Copying the `docker.io/rabbitmq:3-management-alpine` container image list to quay.io via `skopeo copy --all` worked up to  Jul 31, 2023, when it started to fail with `application/vnd.oci.image.index.v1+json(uploading manifest 3-management-alpine to [quay.io/cki/mirror_rabbitmq](http://quay.io/cki/mirror_rabbitmq): manifest invalid), application/vnd.docker.distribution.manifest.list.v2+json(uploading manifest 3-management-alpine to [quay.io/cki/mirror_rabbitmq](http://quay.io/cki/mirror_rabbitmq): manifest invalid)`.

Starting this date, the image list manifest changed to `application/vnd.oci.image.index.v1+json`. Discussing this with the quay.io folks, they were pointing at one of the referenced manifests having the `mediaType` of `application/vnd.docker.distribution.manifest.v2+json` as the most likely reason.

With this, it was possible to workaround this via the `--format oci` parameter of skopeo which converts the manifest above to `application/vnd.oci.image.manifest.v1+json` which makes the copy to go through.

### Reproduction steps

1. get a quay.io account
2. skopeo login quay.io
3. on e.g. a recent Fedora container, run
4. skopeo copy --all docker://docker.io/rabbitmq:3-management-alpine docker://quay.io/foo/bar:baz

---

This had been previously (incorrectly) filed as an issue at https://github.com/rabbitmq/rabbitmq-server/discussions/9133:

> @mh21 please start a discussion here, don't just blindly assume this is a bug

Is this behavior intended, i.e. is this not a bug but a feature of some sort 🤗?

**Comment by yosifkit** [comment]
Ah, looking over the manifest (https://explore.ggcr.dev/?image=rabbitmq%3A3-management-alpine@sha256:dc50fb7c766c9c9c11f8cd1cd05ce2c3960b2c0d7bffe9cce3b3d7e581543fe6), it looks like the `arm32v7` image is still an older classic `docker build` artifact. It has been failing to build since moving to a new pipeline and https://github.com/docker-library/rabbitmq/pull/657 wasn't quite enough to fix it. I'll work on that today.


---

On a side note, it is valid according to the [OCI image index spec](https://github.com/opencontainers/image-spec/blob/v1.1.0-rc4/image-index.md) to have an item in `manifests:` with an "unknown" value:

> v1.1.0-rc4:
> An encountered `mediaType` that is unknown to the implementation MUST NOT generate an error.
>
> v1.0.2:
>An encountered `mediaType` that is unknown SHOULD be safely ignored.

**Reply by lukebakken** [reply]
Whew, am I glad you understand what's going on 😅

**Comment by mh21** [comment]
> On a side note, it is valid according to the [OCI image index spec](https://github.com/opencontainers/image-spec/blob/v1.1.0-rc4/image-index.md) to have an item in `manifests:` with an "unknown" value:

Thank you! I forwarded this to the quay.io folks.

**Comment by mh21** [comment]
Verified that this is now fixed:

```
$ skopeo inspect --raw docker://quay.io/cki/mirror_rabbitmq:3-management-alpine|jq|grep mediaType
  "mediaType": "application/vnd.oci.image.index.v1+json",
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
```
